Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.FileDialog.cs

```diff
@@ -0,0 +1,268 @@
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Collections;
+using System.ComponentModel;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Diagnostics.Contracts;
+
+namespace System.Windows.Forms
+{
+    public abstract class FileDialog // : CommonDialog
+    {
+        // <summary>
+        // Gets or sets a value indicating whether the dialog box automatically adds an extension to a file name if the user omits the extension.
+        // </summary>
+        // 
+        // <returns>
+        // true if the dialog box adds an extension to a file name if the user omits the extension; otherwise, false. The default value is true.
+        // </returns>
+        // public bool AddExtension {get; set;}
+
+        // <summary>
+        // Gets or sets a value indicating whether the dialog box displays a warning if the user specifies a file name that does not exist.
+        // </summary>
+        // 
+        // <returns>
+        // true if the dialog box displays a warning if the user specifies a file name that does not exist; otherwise, false. The default value is false.
+        // </returns>
+        // public virtual bool CheckFileExists {get; set;}
+
+        // <summary>
+        // Gets or sets a value indicating whether the dialog box displays a warning if the user specifies a path that does not exist.
+        // </summary>
+        // 
+        // <returns>
+        // true if the dialog box displays a warning when the user specifies a path that does not exist; otherwise, false. The default value is true.
+        // </returns>
+        // public bool CheckPathExists {get; set;}
+
+        /// <summary>
+        /// Gets or sets the default file name extension.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The default file name extension. The returned string does not include the period. The default value is an empty string ("").
+        /// </returns>
+        public string DefaultExt
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<string>() != null);
+                return default(string);
+            }
+            set
+            {
```

Comment:
I don't think this postcondition makes sense without appropriate precondition in a set method.
The only case this postcondition would be true, if `set` method would not be allowed to set non-null string...